hazelcast/src/main/java/com/hazelcast/map/EntryStore.java

```diff
@@ -38,5 +38,5 @@
  * @param <K> type of the EntryStore key
  * @param <V> type of the EntryStore value
  */
-public interface EntryStore<K, V> extends EntryLoader<K, V>, MapStore<K, EntryLoaderEntry<V>> {
+public interface EntryStore<K, V> extends EntryLoader<K, V>, MapStore<K, ExtendedValue<V>> {
```

Comment:
Why does `EntryStore` extend `EntryLoader`? For convenience?

Reply:
EntryStore extends EntryLoader just as MapStore extends MapLoader. That is because even for plain map operations, MapStore/EntryStore needs to be able to retrieve old values from storage(to return old values). Also other operations such as remove-if-same or replace require loading values from underlying store.

Reply:
Thanks for the clarification, somehow I missed that `MapStore` extends `MapLoader` as well.